tools/build_framework.sh

```diff
@@ -10,7 +10,8 @@ fi
 
 FRAMEWORK_NAME=$1
 FRAMEWORK_DIR=$2
```

Comment:
I'm not sure I understand.  Conditional compilation is accomplished by describing a dependency in the build.gradle file for a framework.  In `hello-world`'s case it wants to be on HEAD all the time so it describes a dependency on the executor project.  On the other hand `kafka` wants to stay on a stable released version of the Executor, so it's resources.json file refers to the published executor and no dependency on the executor project is listed in the build.gradle file.

@mohitsoni has also indicated that as part of the development experience we should probably be moving towards something that relies to as great a degree as possible on gradle and as little as possible on scripts.  @mohitsoni, do you have any thoughts here?  I'll list you as a reviewer.